src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs

```diff
@@ -1160,17 +1155,15 @@ private Expr UnwrapExpression(Expr pExpression)
 
         private static ErrorCode GetStandardLvalueError(CheckLvalueKind kind)
         {
+            Debug.Assert(Enum.IsDefined(typeof(CheckLvalueKind), kind));
             switch (kind)
             {
-                default:
-                    VSFAIL("bad kind");
-                    return ErrorCode.ERR_AssgLvalueExpected;
-                case CheckLvalueKind.Assignment:
-                    return ErrorCode.ERR_AssgLvalueExpected;
                 case CheckLvalueKind.OutParameter:
                     return ErrorCode.ERR_RefLvalueExpected;
                 case CheckLvalueKind.Increment:
                     return ErrorCode.ERR_IncrementLvalueExpected;
+                default:
+                    return ErrorCode.ERR_AssgLvalueExpected;
```

Comment:
But if someone adds a new member of the enum, previously it would hit the assert as it wasn't in the switch statement, prompting us to update this method. After this change it would not hit the assert right?